### PR TITLE
Implement putAllWithMetadata [HZ-2583]

### PR DESCRIPTION
**Pull request by frant-hartm**
Client protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/468

Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java [hunk 1]
```diff
@@ -1667,6 +1695,47 @@ private void putAllInternal(@Nonnull Map<? extends K, ? extends V> map,
...
+                                        .withLastAccessTime(e.getLastAccessTime())
+                                        .withLastStoredTime(e.getLastStoredTime())
+                                        .withLastUpdateTime(e.getLastUpdateTime())
+                                        .withVersion(e.getVersion())
+                                        .withTtl(e.getTtl())
+                                        .withMaxIdle(e.getMaxIdle());
+
+                            }
+                        })
+                       .collect(groupingBy((SimpleEntryView<Data, Data> e) -> {
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java:1722]: Alternative: ".collect(groupingBy((SimpleEntryView<Data, Data> e) -> partitionService.getPartitionId(e.getKey())));"

Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java:1722]: 17743b0392a

File: hazelcast/src/test/java/com/hazelcast/client/map/ClientMapTest.java [hunk 2]
```diff
@@ -380,6 +390,56 @@ public void testAsyncPutAll()  {
...
+
+        SoftAssertions sa = new SoftAssertions();
+
+        sa.assertThat(actual.getCreationTime()).describedAs("creationTime").isEqualTo(entryView.getCreationTime());
+        sa.assertThat(actual.getExpirationTime()).describedAs("expirationTime").isEqualTo(entryView.getExpirationTime());
+        sa.assertThat(actual.getLastAccessTime()).describedAs("lastAccessTime").isEqualTo(entryView.getLastAccessTime());
+        sa.assertThat(actual.getLastUpdateTime()).describedAs("lastUpdateTime").isEqualTo(entryView.getLastUpdateTime());
+        sa.assertThat(actual.getTtl()).describedAs("ttl").isEqualTo(entryView.getTtl());
+        sa.assertThat(actual.getMaxIdle()).describedAs("maxIdle").isEqualTo(entryView.getMaxIdle());
+
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply hazelcast/src/test/java/com/hazelcast/client/map/ClientMapTest.java:426]: 08d64aab53e

File: hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java [hunk 3]
```diff
@@ -1667,6 +1695,47 @@ private void putAllInternal(@Nonnull Map<? extends K, ? extends V> map,
         }
     }
 
+    public CompletableFuture<Void> putAllWithMetadata(@Nonnull Collection<? extends EntryView<K, V>> entries) {
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java]: 483c9a1c8f9

**Review by srknzl — APPROVED** [review]

**Pull request merged**